frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install-wizard-steps/storage-and-nodes-step.tsx

```diff
@@ -75,40 +75,59 @@ export const StorageAndNodes: React.FC<StorageAndNodesProps> = ({ state, dispatc
   const isArbiterSupported = useFlag(GUARDED_FEATURES.OCS_ARBITER);
   const isTaintSupported = useFlag(GUARDED_FEATURES.OCS_TAINT_NODES);
   const [pvData, pvLoaded, pvLoadError] = useK8sWatchResource<K8sResourceKind[]>(pvResource);
-  const [nodesData, nodesLoaded, nodesError] = useK8sWatchResource<NodeKind[]>(nodeResource);
 
   const {
     storageClass,
     storageClassName: scName,
     enableMinimal,
-    nodes,
     stretchClusterChecked,
     enableFlexibleScaling,
+    lvsIsSelectNodes,
+    lvsSelectNodes,
+    lvsAllNodes,
   } = state;
 
-  const scNodeNames: React.MutableRefObject<string[]> = React.useRef([]); // names of the nodes, backing the storage of selected storage class
+  const selectedNodes = lvsIsSelectNodes ? lvsSelectNodes : lvsAllNodes;
 
-  const { cpu, memory, zones } = getNodeInfo(nodes);
-  const nodesCount: number = nodes.length;
+  const memoizedPvData = useDeepCompareMemoize(pvData, true);
+
+  const pvs: K8sResourceKind[] = React.useMemo(() => getSCAvailablePVs(memoizedPvData, scName), [
+    memoizedPvData,
+    scName,
+  ]);
+  const associatedNodes = getAssociatedNodes(pvs);
+
+  const tableData: NodeKindWithLoading[] =
+    associatedNodes?.length > 0 && pvLoaded
+      ? selectedNodes.map((node) => {
+          const nodeName = node.metadata.labels?.['kubernetes.io/hostname'];
+          return associatedNodes?.includes(nodeName) || !state.chartNodes.has(nodeName)
+            ? node
```

Comment:
When selected nodes carrying a node without disk, `associatedNodes?.includes(nodeName)` will be false and loading will always be `true.
Hence added `!state.chartNodes.has(nodeName)`

cc @bipuladh